lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -260,26 +306,88 @@ public static Domain getDomain(
+                    && type instanceof DecimalType
+                    && ((DecimalType) type).isShort())) {
+            List<Long> minimums = converter.getValuesAsLong(type, columnIndex.getMinValues(), columnName);
+            if (!minimums.isEmpty()) {
+                List<Long> maximums = converter.getValuesAsLong(type, columnIndex.getMaxValues(), columnName);
+                if (!maximums.isEmpty()) {
+                    return createDomain(type, columnIndex, id, columnName, hasNullValue, minimums, maximums, (BiFunction<Long, Long, ParquetLongStatistics>) (min, max) -> new ParquetLongStatistics(min, max));
+                }
+            }
+        }
+        else if (descriptor.getType().equals(PrimitiveTypeName.DOUBLE)) {
+            List<Double> minimums = converter.getValuesAsDouble(type, columnIndex.getMinValues(), columnName);
+            if (!minimums.isEmpty()) {
+                List<Double> maximums = converter.getValuesAsDouble(type, columnIndex.getMaxValues(), columnName);
+                if (!maximums.isEmpty()) {
+                    return createDomain(type, columnIndex, id, columnName, hasNullValue, minimums, maximums, (BiFunction<Double, Double, ParquetDoubleStatistics>) (min, max) -> new ParquetDoubleStatistics(min, max));
+                }
+            }
+        }
+        else if (descriptor.getType().equals(PrimitiveTypeName.BINARY)
+                || (descriptor.getType().equals(PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY)
+                    && type instanceof DecimalType
+                    && !((DecimalType) type).isShort())) {
+            List<Slice> minimums = converter.getValuesAsSlice(type, columnIndex.getMinValues());
+            if (!minimums.isEmpty()) {
+                List<Slice> maximums = converter.getValuesAsSlice(type, columnIndex.getMaxValues());
+                if (!maximums.isEmpty()) {
+                    return createDomain(type, columnIndex, id, columnName, hasNullValue, minimums, maximums, (BiFunction<Slice, Slice, ParquetSliceStatistics>) (min, max) -> new ParquetSliceStatistics(min, max));
+                }
             }
-            return Optional.of(new ParquetIntegerStatistics(longStatistics.genericGetMin(), longStatistics.genericGetMax()));
         }
 
-        if (statistics instanceof IntStatistics) {
+        return Domain.create(ValueSet.all(type), hasNullValue);
+    }
+
+    private static ParquetLongStatistics toParquetLongStatistics(Statistics<?> statistics, ParquetDataSourceId id, String column)
+    {
+        if (statistics instanceof LongStatistics) {
+            LongStatistics longStatistics = (LongStatistics) statistics;
+            return new ParquetLongStatistics(longStatistics.genericGetMin(), longStatistics.genericGetMax());
+        }
+        else if (statistics instanceof IntStatistics) {
             IntStatistics intStatistics = (IntStatistics) statistics;
-            if (intStatistics.genericGetMin() > intStatistics.genericGetMax()) {
```

Comment:
This check seems to be gone. Why?

Reply:
This is checked when the range is created deeper in the code.